### ISSUE TITLE
Loading legacy provider breaks SSL

### ISSUE THREAD

**Issue by jeroen** [openssl/openssl#12340]
In the R bindings, we want to enable the legacy provider for now, because some people are still using legacy algorithms.

However after enabling the legacy provider, `SSL_CTX_new` always returns `NULL` (without setting any error). Do I need to enable additional providers to re-enable SSL? 

Below an example of roughly what I am doing, is this incorrect?

 

```c
#include <stdio.h>
#include <openssl/err.h>
#include <openssl/ssl.h>
#include <openssl/provider.h>

static void raise_error(){
  char buf[8192] = {0};
  BIO *bp = BIO_new(BIO_s_mem());
  ERR_print_errors(bp);
  int len = BIO_read(bp, buf, 8192);
  BIO_free(bp);
  printf("OpenSSL error: %.*s", len, buf);
}

static void bail(int success){
  if(!success){
    raise_error();
    exit(1);
  }
}

int main( int argc, const char* argv[] ){
  OSSL_PROVIDER_load(NULL, "legacy"); // problem disappears if you comment this out
  OSSL_PROVIDER_load(NULL, "default");
  OpenSSL_add_all_digests();
  OpenSSL_add_all_algorithms();
  OpenSSL_add_all_ciphers();
  ERR_load_crypto_strings();
  SSL_load_error_strings();
  SSL_library_init();  
  SSL_CTX *ctx = SSL_CTX_new(SSLv23_client_method());
  bail(!!ctx);
  printf("Success!\n");
  return 0;
}
```

**Comment by mattcaswell** [comment]
@jeroen - thanks for your report. I'm closing this as it is a dupe of #12286. There is a fix in #12292.

**Comment by jeroen** [comment]
Thanks! I'll re-run my unit tests on master once that PR gets merged.
